Factory/PredisParametersFactory.php

```diff
@@ -27,7 +27,10 @@ public static function create($options, $class, $dsn)
         $dsnOptions = static::parseDsn(new RedisDsn($dsn));
         $dsnOptions = array_merge($defaultOptions, $options, $dsnOptions);
 
-        if (isset($dsnOptions['persistent'], $dsnOptions['database']) && true === $dsnOptions['persistent']) {
+        if (isset($dsnOptions['persistent'], $dsnOptions['database'])
+            && true === $dsnOptions['persistent']
+            && (int)$dsnOptions['database'] !== 0
```

Comment:
Sorry I don't get it. Why should the persistent option be ignored with database 0 ?

Reply:
Oh I got it now